frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx

```diff
@@ -76,11 +87,28 @@ const VMConsoles: React.FC<VMConsolesProps> = ({
       <VMIsDown />
     );
   }
-
   const vncServiceManual = (vnc && vnc.manual) || undefined;
   const rdpServiceManual = (rdp && rdp.manual) || undefined;
 
-  const desktopViewverSelector = isWindows(vm) ? (
+  const vmName = getName(vm);
+  const namespace = getNamespace(vm);
+  const typeNotSupported =
+    (!showVNCOption && type === ConsoleType.VNC) ||
+    (!showSerialOption && type === ConsoleType.SERIAL);
+
+  const getAvailableType = () => {
+    if (showVNCOption) {
+      return ConsoleType.VNC;
+    }
+    if (showSerialOption) {
+      return ConsoleType.SERIAL;
+    }
+    return null;
+  };
+
+  const consoleType = typeNotSupported || type === undefined ? getAvailableType() : type;
```

Comment:
nit: IMO it is always better to use type == null unless we have a special use case that has different behaviour for null vs undefined

Reply:
fixed